cli/internal/cache/cache.go

```diff
@@ -272,11 +272,12 @@ func (mplex *cacheMultiplexer) Fetch(target string, key string, files []string)
 			// Store this into other caches. We can ignore errors here because we know
 			// we have previously successfully stored in a higher-priority cache, and so the overall
 			// result is a success at fetching. Storing in lower-priority caches is an optimization.
-			_ = mplex.storeUntil(target, key, duration, actualFiles, i)
+			_ = mplex.storeUntil(anchor, key, duration, actualFiles, i)
 			return ok, actualFiles, duration, err
 		}
 	}
-	return false, files, 0, nil
+
+	return false, nil, 0, nil
```

Comment:
With `ok` being false, we should be returning `nil` instead of the input list of `files`.